### ISSUE TITLE
🚀 [Feature] On A Post Show User ID Button In User Hover Menu

### ISSUE THREAD

**Issue by Tirokk** [Human-Connection-Community/Human-Connection-New-20201003#2784]
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-15T10:55:47Z" title="Wednesday, January 15th 2020, 11:55:47 am +01:00">Jan 15, 2020</time>_
_Closed <time datetime="2020-04-23T09:22:04Z" title="Thursday, April 23rd 2020, 11:22:04 am +02:00">Apr 23, 2020</time>_
---

## :rocket: Feature

As a user I like to see the user profile ID of a user commented on a post or somewhere else.
It should show up if I hover over the user slug of a post or a comment and click on a button named `Show all user IDs`.

1. After the click all the **authors** IDs of **the post** and of **all comments** in the list on the post page shall appear. They shall be visible until I refresh the page.
2. The same shall happen if I hover over the post author slug and click in the same button on the menu which appears there. (is decided with @DennisHack 15.01.2020)

Suggestion:

<img width="1813" alt="Bildschirmfoto 2020-01-15 um 11 28 15" src="https://user-images.githubusercontent.com/25344101/72439888-108e0e00-37a8-11ea-9ccb-a949d8a5ff40.png">

### User Problem

If some users change their slugs or names I as a user like to have the ability to see a unique identifier to identify a user safely.

### Implementation

See above. ☝🏼

### Design & Layout

See above. ☝🏼

### Validation

None

### Additional context

None

**Comment by Tirokk** [comment]
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-23T12:14:39Z" title="Thursday, January 23rd 2020, 1:14:39 pm +01:00">Jan 23, 2020</time>_

---


My dear @alina-beck !
How you would design this nicely?
Please contact me personally ;-)

**Comment by Tirokk** [comment]
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-01-27T11:40:43Z" title="Monday, January 27th 2020, 12:40:43 pm +01:00">Jan 27, 2020</time>_

---


This is not so easy!

First off, I don't like the idea of showing all `user IDs` on a page by clicking a button hidden in the hover menu of _one_ user. And then hiding them again by reloading the page – this is not at all transparent and I would strongly advise against it from a UX perspective.

Also, why would I be able to see all `user IDs` on the `Post page` but nowhere else? Maybe I want to see the IDs of all post authors on the `index page` or check the IDs of users following another user, on the `profile page`. Why is that less important? And how do we explain to the users where this feature exists and where it doesn't? (It's also more difficult to implement – we'd have to check for the route to display or hide the button accordingly.)

Another way to implement this would be to have a `show user IDs` button somewhere globally available. I'm not sure where, though. In the `Avatar Menu`, maybe? Or in the `footer`? Either way, it should have the same effect on _all_ pages. It could also be a `setting`, similar to allowing `Embeds` or not.

And there's another issue: displaying the `user IDs` next to (or close to) the user slug will definitely mess up the design – on any page but the `Post page` where there's usually enough space. Except on mobile, of course.

One way to solve this would be to _replace_ the `username` or `date-time` below the slug with the `user ID` when the setting is active – when it's a temporary change I think this solution is preferable to a messed up layout.

@Tirokk @DennisHack what do you think? Let's maybe have a chat about this feature before we start building anything? :wink:

**Comment by Tirokk** [comment]
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-01-27T11:42:16Z" title="Monday, January 27th 2020, 12:42:16 pm +01:00">Jan 27, 2020</time>_

---


I also think there's an issue open – or did we just talk about it in some meeting? – to add the `user ID` to the `avatar hover menu`, so when hovering over an avatar you could see the `ID` of this user in the popup.

**Comment by Tirokk** [comment]
> [<img alt="stale[bot]" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/apps/stale) **Authored by [stale[bot]](https://github.com/apps/stale)**
_<time datetime="2020-04-03T13:04:39Z" title="Friday, April 3rd 2020, 3:04:39 pm +02:00">Apr 3, 2020</time>_

---


This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by Tirokk** [comment]
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-04-23T05:57:42Z" title="Thursday, April 23rd 2020, 7:57:42 am +02:00">Apr 23, 2020</time>_

---


Not stale!

**Comment by Tirokk** [comment]
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-23T07:30:40Z" title="Thursday, April 23rd 2020, 9:30:40 am +02:00">Apr 23, 2020</time>_

---


#3093 removed the hover card from user-teaser, didn't it?

Nevermind; I found #3007.
Can't we just close this one, then?

**Comment by Tirokk** [comment]
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-04-23T09:22:04Z" title="Thursday, April 23rd 2020, 11:22:04 am +02:00">Apr 23, 2020</time>_

---


Yes, you are right @rbeer ! 👍🏼

We go for #3007 and I close this now.
